### PR TITLE
Requested changes from featured branch.

### PR DESCRIPTION
**Pull request by chernecov**
@RuslanZavacky  Please review changes, I've moved them from "featured" branch.

### CODE REVIEW

File: Configuration/Resource.php [hunk 1]
```diff
@@ -27,16 +27,28 @@ class Resource
...
      */
-    public function __construct($type, $showLinkSelf = null)
+    protected $absolute = false;
+
+    /**
+     * Resource constructor
+     *
+     * @param string    $type
+     * @param bool|null $showLinkSelf
+     * @param bool|     $absolute
```
Comment (RuslanZavacky) [inline Configuration/Resource.php]: `bool|null`

File: EventListener/Serializer/JsonEventSubscriber.php [hunk 2]
```diff
@@ -214,10 +213,19 @@ public function onPostSerialize(ObjectEvent $event)
...
-            ));
+        /** @var Relationship $resource */
+        $resource = $metadata->getResource();
+        if ($resource && true === $resource->getShowLinkSelf()) {
+            $uri = $this->baseUriResolver->getBaseUri($resource->isAbsolute());
+            $visitor->addData(
+                'links',
+                [
+                    'self' =>
+                        $uri . '/' .
```
Comment (RuslanZavacky) [inline EventListener/Serializer/JsonEventSubscriber.php]: Maybe its ok to actually have it in one line?

**Comment by RuslanZavacky** [comment]
Nitpick comments, apart from that, all good 👍

File: Configuration/Resource.php (hunk 1 above)
Comment (chernecov) [reply Configuration/Resource.php]: Done.

File: EventListener/Serializer/JsonEventSubscriber.php (hunk 2 above)
Comment (chernecov) [reply EventListener/Serializer/JsonEventSubscriber.php]: Done.

**Pull request merged**